features/filesystem/fat/FATFileSystem.h

```diff
@@ -49,18 +49,18 @@ class FATFileSystem : public FileSystem {
      *  The block device to format should be mounted when this function is called.
      *
      *  @param bd
-     *    This is the block device that will be formated.
-     *
-     *  @param allocation_unit
-     *    This is the number of bytes per cluster size. The valid value is N
-     *    times the sector size. N is a power of 2 from 1 to 128 for FAT
-     *    volume and upto 16MiB for exFAT volume. If zero is given,
-     *    the default allocation unit size is selected by the underlying
-     *    filesystem, which depends on the volume size.
+     *    This is the block device that will be formatted.
+     *
+     *  @param cluster_size
+     *    This is the number of bytes per cluster. A larger cluster size will decrease
+     *    the overhead of the FAT table, but also increase the minimum file size. The
+     *    cluster_size must be a multiple of the nearest power-of-2 of the underlying
+     *    device's erase_size. If zero, a cluster size will be determined from the
```

Comment:
Device erase_size can be as less as single block. Cluster size (nearest power of 2 of erase_size 512bytes) will be 1024 bytes and not good value to set. 

Cluster size is logical  boundary in files system.

AU size on SD cards is not logical boundary, but physical boundary in user area of card, and formatting card with correct AU size gives optimal performance. Setting cluster size equivalent to AU size is recommended for SD cards.

Reply:
Ah, I changed this over here 
https://github.com/ARMmbed/mbed-os/pull/4843/files#diff-d2c0be70be00c3c1688a6e00f622217bR54

I just haven't rebased this pr yet